examples/src/main/scala/App.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import com.microsoft.hyperspace._
+import com.microsoft.hyperspace.Hyperspace
+import com.microsoft.hyperspace.index.{IndexConfig, IndexConstants}
+
+object App {
+  def main(args: Array[String]): Unit = {
+    // Create Spark session
+    val sparkConf = new SparkConf().setMaster("local[*]")
+    val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+    spark.sparkContext.setLogLevel("ERROR")
+
+    // Sample department records
+    val departments = Seq(
+      (10, "Accounting", "New York"),
+      (20, "Research", "Dallas"),
+      (30, "Sales", "Chicago"),
+      (40, "Operations", "Boston"))
+
+    // Sample employee records
+    val employees = Seq(
+      (7369, "SMITH", 20),
+      (7499, "ALLEN", 30),
+      (7521, "WARD", 30),
+      (7566, "JONES", 20),
+      (7698, "BLAKE", 30),
+      (7782, "CLARK", 10),
+      (7788, "SCOTT", 20),
+      (7839, "KING", 10),
+      (7844, "TURNER", 30),
+      (7876, "ADAMS", 20),
+      (7900, "JAMES", 30),
+      (7934, "MILLER", 10),
+      (7902, "FORD", 20),
+      (7654, "MARTIN", 30))
+
+    // Save example data records as Parquet
+    import spark.implicits._
+    val deptLocation = "departments"
+    val empLocation = "employees"
+    departments
+      .toDF("deptId", "deptName", "location")
+      .write
+      .mode("overwrite")
+      .parquet(deptLocation)
+
+    employees
+      .toDF("empId", "empName", "deptId")
+      .write
+      .mode("overwrite")
+      .parquet(empLocation)
+
+    // Create Hyperspace indexes
+    val hyperspace = new Hyperspace(spark)
+
+    val empDF: DataFrame = spark.read.parquet(empLocation)
+    val deptDF: DataFrame = spark.read.parquet(deptLocation)
+
+    val empIndexConfig: IndexConfig = IndexConfig("empIndex", Seq("deptId"), Seq("empName"))
+    val deptIndexConfig: IndexConfig = IndexConfig("deptIndex", Seq("deptId"), Seq("deptName"))
+    hyperspace.createIndex(empDF, empIndexConfig)
+    hyperspace.createIndex(deptDF, deptIndexConfig)
+
+    // List all indexes
+    hyperspace.indexes.show
+
+    // Enable Hyperspace to leverage indexes
+    spark.enableHyperspace()
+    val deptDFrame: DataFrame = spark.read.parquet("departments")
+    val empDFrame: DataFrame = spark.read.parquet(empLocation)
```

Comment:
I'm curious. Why did you define `empDFrame` and `deptDFrame` as opposed to just using the `empDF` and `deptDF` above?

Reply:
It is just to show that index usage has nothing to do with which DataFrame the query uses (i.e. For a query, user does not need to use the same DF he used for indexing). We follow the same pattern in our notebook example. Plz let me know if you prefer to drop these two DFs and directly write queries against original DFs used to create indexes.

Reply:
I would probably simplify it - since there are no comments anyways to clarify what you mentioned.

Reply:
@rapoth I removed new DFs and used same DFs for both creating indexes and queries.